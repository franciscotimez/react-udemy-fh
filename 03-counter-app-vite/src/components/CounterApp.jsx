import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ initValue }) => {

  const [value, setValue] = useState(initValue);

  const handleAdd = () => {
    // setValue(value + 1);
    setValue( (v) => v + 1 );
  };

  const handleSub = () => {
    setValue( (v) => v - 1 );
  };

  const handleReset = () => {
    setValue( initValue );
  };


  return (
    <>
      <h1>CounterApp</h1>
      <h2> {value} </h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleSub}>-1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};


CounterApp.propTypes = {
  initValue: PropTypes.number.isRequired
};