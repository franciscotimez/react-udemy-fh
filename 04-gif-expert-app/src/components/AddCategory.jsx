import { useState } from "react";

export const AddCategory = ({onAddCategory}) => {
  const [inputValue, setInputValue] = useState('');

  const handleChange = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    const newCategory = inputValue.trim()
    
    if(newCategory.length > 1) {
      onAddCategory(newCategory)
    }
    setInputValue('')
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        type='text'
        placeholder='Buscar Gifs'
        onChange={handleChange}
        value={inputValue}
      />
    </form>
  );
};
