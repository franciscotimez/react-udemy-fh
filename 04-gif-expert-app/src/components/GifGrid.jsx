

export const GifGrid = ({category}) => {
  return (
    <div>{category}</div>
  )
}
