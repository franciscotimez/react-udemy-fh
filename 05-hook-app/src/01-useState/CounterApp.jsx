import { useState } from "react";


export const CounterApp = () => {

  const [counter, setCounter] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30,
  });

  const {counter1, counter2, counter3} = counter;

  const onHandleClick = ({target}) => {
    console.log(event);
    setCounter({
      ...counter,
      [target.name]: counter[target.name] + 1
    })
  }

  return (
    <>
      <h1>Counter1: {counter1}</h1>
      <button
        className="btn btn-primary"
        name="counter1"
        onClick={onHandleClick}
      >
        +1
      </button>
      <hr />
      <h1>Counter2: {counter2}</h1>
      <button
        className="btn btn-primary"
        name="counter2"
        onClick={onHandleClick}
      >
        +1
      </button>
      <hr />
      <h1>Counter3: {counter3}</h1>
      <button
        className="btn btn-primary"
        name="counter3"
        onClick={onHandleClick}
      >
        +1
      </button>
      <hr />
    </>
  );
};
