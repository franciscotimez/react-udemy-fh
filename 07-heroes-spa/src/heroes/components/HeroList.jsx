import { getHeroesByPublisher } from "../helpers/getHeroesByPublisher";
import { HeroCard } from "./HeroCard";


export const HeroList = ({ publisher }) => {

  const heroes = getHeroesByPublisher(publisher);

  return (
    <div className="row row-cols-1 row-cols-md-3 g-3">
      {heroes.map((hero) => (
        <HeroCard key={hero.id} {...hero} />
      ))}
    </div>
  );
};
