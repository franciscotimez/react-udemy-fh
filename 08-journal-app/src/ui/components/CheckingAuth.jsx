import { CircularProgress, Grid } from "@mui/material";

export const CheckingAuth = () => {
  return (
    <Grid
      container
      spacing={0}
      direction="column"
      justifyContent="center"
      alignItems="center"
      sx={{ minHeight: '100vh', backgroundColor: 'primary.main', padding: 4 }}
    >
      <Grid
        item
        justifyContent="center"
      >
        <CircularProgress color="warning" />
      </Grid>
    </Grid>
  );
};
