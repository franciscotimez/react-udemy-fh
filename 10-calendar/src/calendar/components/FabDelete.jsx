import { useCalendarStore, useUiStore } from "../../hooks";

export const FabDelete = () => {

  const { startDeletingEvent, hasEventSelected } = useCalendarStore();

  const handleClickDelete = () => {
    startDeletingEvent();
  };

  return (
    <button
      aria-label="btn-delete"
      className="btn btn-danger fab-danger"
      onClick={handleClickDelete}
      style={{
        display: hasEventSelected ? '' : "none"
      }}
    >
      <i className="fas fa-trash-alt" ></i>
    </button>
  );
};
