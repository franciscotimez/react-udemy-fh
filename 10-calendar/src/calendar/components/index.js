export * from './CalendarEvent'
export * from './NavBar'
export * from './CalendarModal'
export * from './FabAddNew'
export * from './FabDelete'