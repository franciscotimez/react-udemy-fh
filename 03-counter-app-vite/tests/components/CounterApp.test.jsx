import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../../src/components/CounterApp";


describe('Test en el componente <CounterApp />', () => {

    const initValue = 100;

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<CounterApp initValue={initValue} />);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar el valor inicial', () => {
        render(<CounterApp initValue={initValue} />);
        expect(screen.getByText(initValue)).toBeTruthy();
    });

    test('debe incrementar el contador con el boton +1', () => {
        render(<CounterApp initValue={initValue} />);
        fireEvent.click( screen.getByText('+1') );
        expect(screen.getByText(initValue + 1)).toBeTruthy();
    });

    test('debe decrementar el contador con el boton -1', () => {
        render(<CounterApp initValue={initValue} />);
        fireEvent.click( screen.getByText('-1') );
        expect(screen.getByText(initValue - 1)).toBeTruthy();
    });

    test('debe funcionar el boton de reset', () => {
        render(<CounterApp initValue={initValue} />);
        const incBtn = screen.getByText('+1')
        fireEvent.click( incBtn );
        fireEvent.click( incBtn );
        fireEvent.click( incBtn );

        fireEvent.click( screen.getByRole('button', { name: 'btn-reset' }) );

        expect(screen.getByText(initValue)).toBeTruthy();
    });
});