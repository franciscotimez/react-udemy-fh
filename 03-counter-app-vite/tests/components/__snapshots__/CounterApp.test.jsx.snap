// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en el componente <CounterApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
